### PR TITLE
Add `xbuildenv create` CLI command and remove old entrypoints

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Add a private `pyodide xbuildenv create` CLI commad that replaces `pyodide-build create_xbuildenv`.  Also removes legacy entrypoints: `pyodide-build {create,install}_xbuildenv.

### Checklists

- [x] Add / update tests
- [x] check CI artifact
- [x] check tests run in CI

### CODE REVIEW

File: pyodide-build/pyodide_build/create_xbuildenv.py [hunk 1]
```diff
@@ -13,44 +12,41 @@
...
     for recipe in recipes.values():
         xbuild_files = recipe.build.cross_build_files
         for path in xbuild_files:
+            source = site_packages / path
             target = site_packages_extras / path
             target.parent.mkdir(parents=True, exist_ok=True)
-            shutil.copy(site_packages / path, target)
 
+            if not source.exists():
+                if skip_missing_cross_file:
```
Comment (ryanking13) [inline pyodide-build/pyodide_build/create_xbuildenv.py:34]: This is mostly for tests. I don't want to build scipy and numpy to test this functionality

File: pyodide-build/pyodide_build/tests/test_cli.py [hunk 2]
```diff
@@ -344,6 +344,39 @@ def test_create_zipfile_compile(temp_python_lib, temp_python_lib2, tmp_path):
...
+        ],
+    )
+    assert result.exit_code == 0, result.stdout
+    assert "xbuildenv created at" in result.stdout
+    assert (envpath / "xbuildenv").exists()
+    assert (envpath / "xbuildenv" / "pyodide-root").is_dir()
+    assert (envpath / "xbuildenv" / "site-packages-extras").is_dir()
+    assert (envpath / "xbuildenv" / "requirements.txt").exists()
+
+    if not package_is_built("scipy"):
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/tests/test_cli.py:367]: Would be nice to mock this, but it's probably too much trouble.

File: pyodide-build/pyodide_build/create_xbuildenv.py [hunk 3]
```diff
@@ -86,14 +91,24 @@ def copy_wasm_libs(xbuildenv_path: Path) -> None:
...
+def create(
+    path: str | Path,
+    pyodide_root: Path | None = None,
+    *,
+    skip_missing_files: bool = False,
+) -> None:
+    if pyodide_root is None:
+        pyodide_root = get_pyodide_root()
+
+    xbuildenv_path = Path(path) / "xbuildenv"
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/create_xbuildenv.py:103]: We should start putting the Pyodide version into these paths!

**Review by hoodmane — APPROVED** [review]

File: pyodide-build/pyodide_build/create_xbuildenv.py (hunk 3 above)
Comment (ryanking13) [reply pyodide-build/pyodide_build/create_xbuildenv.py:103]: Yes, indeed :) We also need to fix hardcoded xbuildenv paths.

**Comment by hoodmane** [comment]
Thanks @ryanking13!

**Pull request merged**